contracts/src/contracts/ValidatorManager/BalancerValidatorManager.sol

```diff
@@ -73,17 +74,14 @@ contract BalancerValidatorManager is
         _;
     }
 
-    constructor(
-        ICMInitializable init
-    ) {
-        if (init == ICMInitializable.Disallowed) {
-            _disableInitializers();
-        }
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
     }
 
     function initialize(
         BalancerValidatorManagerSettings calldata settings
-    ) external initializer {
+    ) external reinitializer(2) {
```

Comment:
Is there a context where we'll need to reinitialize `BalancerValidatorManager`?

Reply:
We always need to reinitialize it as it will always be upgraded from a `PoAValidatorManager`. The same thing is done in `teleporter` for `PoSValidatorManager`s (e.g. `ERC20TokenStakingManager`):

https://github.com/ava-labs/teleporter/blob/179871ccf8f4e6e4633fa524bc75934977ed8347/contracts/validator-manager/ERC20TokenStakingManager.sol#L73